test/unittest/statistics/rtps/RTPSStatisticsTests.cpp

```diff
@@ -0,0 +1,225 @@
+        auto nolistener = listener1;
+        nolistener.reset();
+
+        fastdds::statistics::EventKind kind =
+                fastdds::statistics::EventKind::PUBLICATION_THROUGHPUT;
+        fastdds::statistics::EventKind another_kind =
+                fastdds::statistics::EventKind::SUBSCRIPTION_THROUGHPUT;
+        fastdds::statistics::EventKind yet_another_kind =
+                fastdds::statistics::EventKind::NETWORK_LATENCY;
+
+        // test the participant apis
+        // + fails if no listener has been yet added
+        EXPECT_FALSE(participant->remove_statistics_listener(listener1, kind));
+        // + fails to add an empty listener
+        EXPECT_FALSE(participant->add_statistics_listener(nolistener, kind));
+        // + succeeds to add a new listener
+        ASSERT_TRUE(participant->add_statistics_listener(listener1, kind));
+        // + fails to add multiple times the same listener...
+        EXPECT_FALSE(participant->add_statistics_listener(listener1, kind));
+        //   ... unless it's associated to other entity
+        EXPECT_TRUE(participant->add_statistics_listener(listener1, another_kind));
+        // + fails if an unknown listener is removed
+        EXPECT_FALSE(participant->remove_statistics_listener(listener2, kind));
+        // + fails if a known listener is removed with a non registered entity
+        EXPECT_FALSE(participant->remove_statistics_listener(listener1, yet_another_kind));
+        // + succeeds to remove a known listener
+        EXPECT_TRUE(participant->remove_statistics_listener(listener1, kind));
+        // + fails if a listener is already removed
+        EXPECT_FALSE(participant->remove_statistics_listener(listener1, kind));
+
+        // test the writer apis
+        // + fails if no listener has been yet added
+        EXPECT_FALSE(writer->remove_statistics_listener(listener1));
+        // + fails to add an empty listener
+        EXPECT_FALSE(writer->add_statistics_listener(nolistener));
+        // + succeeds to add a new listener
+        ASSERT_TRUE(writer->add_statistics_listener(listener1));
+        // + fails to add multiple times the same listener
+        EXPECT_FALSE(writer->add_statistics_listener(listener1));
+        // + fails if an unknown listener is removed
+        EXPECT_FALSE(writer->remove_statistics_listener(listener2));
+        // + succeeds to remove a known listener
+        EXPECT_TRUE(writer->remove_statistics_listener(listener1));
+        // + fails if a listener is already removed
+        EXPECT_FALSE(writer->remove_statistics_listener(listener1));
+
+        // test the reader apis
+        // + fails if no listener has been yet added
+        EXPECT_FALSE(reader->remove_statistics_listener(listener1));
+        // + fails to add an empty listener
+        EXPECT_FALSE(reader->add_statistics_listener(nolistener));
+        // + succeeds to add a new listener
+        ASSERT_TRUE(reader->add_statistics_listener(listener1));
```

Comment:
```suggestion
        EXPECT_TRUE(reader->add_statistics_listener(listener1));
```

Reply:
This one must be an assertion because the following expected results rely on that to be true.